app/src/features/dataset_explorer/DatasetExplorer.tsx

```diff
@@ -1,76 +1,61 @@
-import React, { useState, useEffect } from "react";
+import React, { useState } from "react";
 import DatasetFilter from "./DatasetFilter";
-import DataFetcher from "../../utils/DataFetcher";
 import DataTable from "./DataTable";
 import DatasetListing from "./DatasetListing";
 import styles from "./DatasetExplorer.module.scss";
 import ChevronRightIcon from "@material-ui/icons/ChevronRight";
-import { DatasetMetadata } from "../../utils/DatasetMetadata";
-
-type LoadStatus = "unloaded" | "loading" | "loaded";
+import { useDatasetStore } from "../../utils/useDatasets";
+import { DatasetStore } from "../../utils/DatasetTypes";
 
 function renderTableOrPlaceholder(
-  loadStatus: LoadStatus,
-  columns: any[],
-  data: any[]
+  datasetStore: DatasetStore,
+  datasetId: string
 ) {
+  const loadStatus = datasetStore.getDatasetLoadStatus(datasetId);
```

Comment:
can this variable go directly in the switch?

Reply:
Yup